tests/Unit/DataStructures/Tensor/Expressions/Test_Contract.cpp

```diff
@@ -36,20 +36,24 @@ void create_tensor(gsl::not_null<Tensor<DataVector, Ts...>*> tensor) noexcept {
 template <typename DataType>
 void test_contractions_rank2(const DataType& used_for_size) noexcept {
   // Contract (upper, lower) tensor
-  // Use explicit type (vs auto) so the compiler checks the return type of
-  // `evaluate`
+  // Use explicit type (vs auto) for LHS Tensor so the compiler checks the
+  // return type of `evaluate`
   Tensor<DataType, Symmetry<2, 1>,
          index_list<SpatialIndex<3, UpLo::Up, Frame::Inertial>,
                     SpatialIndex<3, UpLo::Lo, Frame::Inertial>>>
       Rul(used_for_size);
   create_tensor(make_not_null(&Rul));
 
-  const Tensor<DataType> RIi_contracted =
-      TensorExpressions::evaluate(Rul(ti_I, ti_i));
+  const auto RIi_expr = Rul(ti_I, ti_i);
+  const Tensor<DataType> RIi_contracted = TensorExpressions::evaluate(RIi_expr);
 
   DataType expected_RIi_sum = make_with_value<DataType>(used_for_size, 0.0);
   for (size_t i = 0; i < 3; i++) {
     expected_RIi_sum += Rul.get(i, i);
+
+    const std::array<size_t, 2> uncontracted_lhs_tensor_index{{i, i}};
```

Comment:
If I correctly understand this is the "expected" value for the CHECK below, then I suggested renaming to `expected_ uncontracted_lhs_tensor_index`, to clarify this

Same comment on all instances of this pattern below